src/mmg3d/hash_3d.c

```diff
@@ -1370,18 +1370,31 @@ int MMG5_bdryTria(MMG5_pMesh mesh, MMG5_int ntmesh) {
          * mesh->nt=0 at the beginning of the function) */
         ptt->cc = 4*k + i;
 
+        /* If in LS mode, in analysis after ls discretization: xtetra exists. It has been created by MMG5_bdrySet */
         if ( pxt ) {
-          /* useful only when saving mesh or in ls mode */
+          /* Useful only when saving mesh or in ls mode */
           for( j = 0; j < 3; j++ ) {
+            /* Assign tags to tria from xtetra->tag and remove redundant boundary tag */
```

Comment:
I would add some additionnal comments to illustrate the issue we found when analysing for the second time the surface intersections in ParMmg : 

/* Assign tags to tria from xtetra->tag and remove redundant boundary tag: when called from ParMmg in ls mode, it is needed to remove the parallel tags coming from previous surface analysis  to ensure the suitable setting of the `MG_BDY` tag along edges at the intersection between geometrical (true) boundaries and purely parallel interfaces. For that, it is mandatory to remove the MG_PARBDYBDY tag already added along such edges (see the step 2 of the mmgHashTria implementation)  */